salt/netapi/rest_cherrypy/app.py

```diff
@@ -1427,6 +1448,14 @@ def POST(self, **kwargs):
         else:
             creds = cherrypy.serving.request.lowstate
 
+        # Authorization header
+        remote_user = cherrypy.request.headers.get('X-Remote-User')
+        sharedsecret =  cherrypy.request.headers.get('X-Shared-Secret')
+        if remote_user and sharedsecret:
+            creds['username'] = username
+            creds['sharedsecret'] = sharedsecret
+            creds['eauth'] = 'sharedsecret'
+
```

Comment:
I would prefer to not pull this in to `rest_cherrypy` because it is a custom solution for a custom problem and not applicable to the wider Salt community. I would feel differently if this was a standard (HTTP basic auth or digest auth), or a widely-adopted non-standard.

This functionality would better live elsewhere. Possibly as a custom netapi module that wraps this module. Or in custom functionality in the front-facing web server -- one that translates custom headers into regular Salt POST data.